google-codelab/google_codelab.js

```diff
@@ -218,6 +262,90 @@ class Codelab extends HTMLElement {
     this.eventHandler_.listen(window, events.EventType.POPSTATE, (e) => {
       this.handlePopStateChanged_(e);
     });
+
+    this.eventHandler_.listen(document.body, events.EventType.KEYDOWN, (e) => {
+      this.handleKeyDown_(e);
+    });
+  }
+
+  /**
+   * @private
+   */
+  toggleToolbar_() {
+    if (!this.titleContainer_) {
+      return;
+    }
+
+    if (this.hasAttribute(NO_TOOLBAR_ATTR)) {
+      this.titleContainer_.setAttribute(HIDDEN_ATTR, '');
+    } else {
+      this.titleContainer_.removeAttribute(HIDDEN_ATTR);
+    }
+  }
+
+  /**
+   * @private
+   */
+  toggleArrows_() {
+    if (!this.controls_) {
+      return;
+    }
+
+    if (this.hasAttribute(NO_ARROWS_ATTR)) {
+      this.controls_.setAttribute(HIDDEN_ATTR, '');
+    } else {
+      this.controls_.removeAttribute(HIDDEN_ATTR);
+    }
+  }
+
+  /**
+   * 
+   * @param {!events.BrowserEvent} e 
+   */
+  handleDrawerKeyDown_(e) {
+    if (!this.drawer_) {
+      return;
+    }
+
+    const focused = this.drawer_.querySelector(':focus');
+    let li;
+    if (focused) {
+      li = /** @type {!Element} */ (focused.parentNode);
+    } else {
+      li = this.drawer_.querySelector(`[${SELECTED_ATTR}]`);
+    }
+
+    if (!li) {
+      return;
+    }
+
+    let next;
+    if (e.keyCode == KeyCodes.UP) {
+      next = dom.getPreviousElementSibling(li);
+    } else if (e.keyCode == KeyCodes.DOWN) {
+      next = dom.getNextElementSibling(li);
+    }
+
+    if (next) {
+      const a = next.querySelector('a');
+      if (a) {
+        a.focus();
+      }
+    }
+  }
+
+  /**
+   * 
+   * @param {!events.BrowserEvent} e 
+   */
+  handleKeyDown_(e) {
+    if (e.keyCode == KeyCodes.LEFT) {
+      document.activeElement.blur();
+      this.selectPrevious();
```

Comment:
Not sure but can `document.activeElement` be `null`? In which case the next `this.selectXxx` won't execute due to thrown error, right.

Reply:
Added a check